src/com/sap/piper/DefaultValueCache.groovy

```diff
@@ -51,13 +60,48 @@ class DefaultValueCache implements Serializable {
             if(customDefaults in List)
                 configFileList += customDefaults
             for (def configFileName : configFileList){
-                if(configFileList.size() > 1) steps.echo "Loading configuration file '${configFileName}'"
-                def configuration = steps.readYaml text: steps.libraryResource(configFileName)
+                if(configFileList.size() > 1) script.echo "Loading configuration file '${configFileName}'"
+                def configuration = script.readYaml text: script.libraryResource(configFileName)
                 defaultValues = MapUtils.merge(
                         MapUtils.pruneNulls(defaultValues),
                         MapUtils.pruneNulls(configuration))
             }
             DefaultValueCache.createInstance(defaultValues, customDefaults)
+
+            if (script) {
+                persistDefaults(script, defaultValues, customDefaults)
+            }
+
+        }
+    }
+
+    static void persistDefaults(Script script, Map defaultValues, List customDefaults = []) {
+        def defaultValuesAbsolutePath = "${script.WORKSPACE}/${defaultValuesRelativePath}"
+        def customDefaultsAbsolutePath = "${script.WORKSPACE}/${customDefaultsRelativePath}"
+
+        if (defaultValues && !script.fileExists(defaultValuesAbsolutePath)) {
```

Comment:
The `fileExists` needs a node context. Unfortunately, the creation of the configuration is done outside the node context of the [piperStageWrapper](https://github.com/SAP/jenkins-library/blob/1b6781c1e9c133d83146ed2a2156b981af96a5d7/vars/piperStageWrapper.groovy#L42). Therefor, the piperPipeline fails:

```
ERROR: Attempted to execute a step that requires a node context while ‘agent none’ was specified. Be sure to specify your own ‘node { ... }’ blocks when using ‘agent none’.
Finished: FAILURE
```